opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/PlatformResources.java

```diff
@@ -33,113 +35,380 @@
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
+import decodes.db.DatabaseException;
+import decodes.db.DatabaseIO;
+import decodes.db.Platform;
+import decodes.db.PlatformConfig;
+import decodes.db.PlatformList;
+import decodes.db.PlatformSensor;
+import decodes.db.PlatformStatus;
+import decodes.db.Site;
+import decodes.db.TransportMedium;
+import decodes.sql.DbKey;
+import decodes.tsdb.DbIoException;
+import opendcs.dai.PlatformStatusDAI;
 import org.opendcs.odcsapi.beans.ApiPlatform;
 import org.opendcs.odcsapi.beans.ApiPlatformRef;
-import org.opendcs.odcsapi.dao.ApiPlatformDAO;
+import org.opendcs.odcsapi.beans.ApiPlatformSensor;
+import org.opendcs.odcsapi.beans.ApiPlatformStatus;
+import org.opendcs.odcsapi.beans.ApiTransportMedium;
 import org.opendcs.odcsapi.dao.DbException;
-import org.opendcs.odcsapi.errorhandling.ErrorCodes;
 import org.opendcs.odcsapi.errorhandling.WebAppException;
-import org.opendcs.odcsapi.hydrojson.DbInterface;
 import org.opendcs.odcsapi.sec.AuthorizationCheck;
-import org.opendcs.odcsapi.util.ApiConstants;
-import org.opendcs.odcsapi.util.ApiHttpUtil;
 
 @Path("/")
-public class PlatformResources
+public class PlatformResources extends OpenDcsResource
 {
 	@Context HttpHeaders httpHeaders;
 
 	@GET
 	@Path("platformrefs")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
-	public Response gePlatformRefs(@QueryParam("tmtype") String tmtype)
-		throws DbException, SQLException
+	public Response getPlatformRefs(@QueryParam("tmtype") String tmtype)
+			throws DbException
 	{
 		HashMap<String, ApiPlatformRef> ret = new HashMap<>();
```

Comment:
HashMap vs Map ?  Is this driven by DTO?   I'm always suspicious of HashMap.  IMO the extra cost of LinkedHashMap is always worth it to maintain insertion order, even if its just to make debugging consistent.

Reply:
I've swapped HashMap for Map and the implementation for LinkedHashMap. I've also replaced ArraList with List where possible/appropriate.